x/auth/vesting/client/cli/tx.go

```diff
@@ -86,3 +89,81 @@ timestamp.`,
 
 	return cmd
 }
+
+type InputPeriod struct {
+	Coins string `json:"coins"`
+	Time  int64  `json:"unix_time"`
+}
+
+// NewMsgCreateVestingAccountCmd returns a CLI command handler for creating a
+// MsgCreateVestingAccount transaction.
+func NewMsgCreatePeriodicVestingAccountCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create-periodic-vesting-account [to_address] [periods_json_file]",
+		Short: "Create a new vesting account funded with an allocation of tokens.",
+		Long: `Create a new vesting account funded with an allocation of tokens. This takes a destingation address and 
+		a period json file. 
+		Where periods.json contains:
+
+		An array of coin strings and unix epoch times for coins to vest
+
+[
+{
+  "coins": "10test",
+  "unix_time":1620016673
+ },
+ {
+	"coins": "10test",
+	"unix_time":1620026673
+   },
+]
+		`,
+		Args: cobra.ExactArgs(3),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			toAddr, err := sdk.AccAddressFromBech32(args[0])
+			if err != nil {
+				return err
+			}
+
+			contents, err := ioutil.ReadFile(args[1])
+			if err != nil {
+				return err
+			}
+
+			var inputPeriods []InputPeriod
+
+			err = json.Unmarshal(contents, inputPeriods)
+			if err != nil {
+				return err
+			}
+
+			var periods []types.Period
+
+			for _, p := range inputPeriods {
+
+				amount, err := sdk.ParseCoinsNormalized(p.Coins)
+				if err != nil {
+					return err
+				}
+				period := types.Period{Length: p.Time, Amount: amount}
```

Comment:
There seems to be a mismatch between `p.Time` which is, as far as I can tell, a UNIX epoch timestamp, and the `Length` field, which `vesting.proto` implies is a duration, and is used in the context of a `start_time` field. It sure looks like this code puts a timestamp value into a duration field. If that's not the case, then either some variables or fields should be renamed, or a big comment added.